src/slack/operations.go

```diff
@@ -103,3 +105,116 @@ func (p *Provider) ShowSearchResultModal(_ context.Context, triggerID, imageURL,
 
 	return nil
 }
+
+// ShowStickerPreview sends a sticker preview to the specified user as an ephemeral message in the Slack channel
+func (p *Provider) ShowStickerPreview(_ context.Context, userID, channelID, tag, imageURL string) error {
+	log := p.logger.With().Str(helper.LogStrKeyMethod, "ShowStickerPreview").Logger()
+	sticker := model.StickerBlockActionValue{
+		Tag:    tag,
+		Index:  0,
+		ImgURL: imageURL,
+	}
+
+	blocks := createStickerPreviewBlock(sticker, true)
+
+	_, _, err := p.client.PostMessage(
+		channelID,
+		slack.MsgOptionPostEphemeral(userID),
+		slack.MsgOptionBlocks(blocks.BlockSet...),
+	)
+	if err != nil {
+		log.Err(err).Msg("PostMessage failed")
+		return err
+	}
+
+	return nil
+}
+
+// ShuffleStickerPreview updates the sticker preview by replacing the original ephemeral message with a new one
+// containing a shuffled sticker, updating the displayed image based on the tag and index.
+func (p *Provider) ShuffleStickerPreview(_ context.Context, userID, channelID, responseURL string, sticker model.StickerBlockActionValue) error {
+	log := p.logger.With().Str(helper.LogStrKeyMethod, "ShuffleStickerPreview").Logger()
+
+	block := createStickerPreviewBlock(sticker, true)
+
+	_, _, err := p.client.PostMessage(
+		channelID,
+		slack.MsgOptionAsUser(true),
+		slack.MsgOptionPostEphemeral(userID),
+		slack.MsgOptionReplaceOriginal(responseURL),
+		slack.MsgOptionBlocks(block.BlockSet...),
+	)
+	if err != nil {
+		log.Err(err).Msg("PostMessage failed")
+		return err
+	}
+
+	return nil
+}
+
+// CancelStickerPreview removes the sticker preview message from Slack by deleting the original ephemeral message.
+func (p *Provider) CancelStickerPreview(_ context.Context, channelID, responseURL string) error {
+	log := p.logger.With().Str(helper.LogStrKeyMethod, "CancelStickerPreview").Logger()
+	_, _, err := p.client.PostMessage(
+		channelID,
+		slack.MsgOptionDeleteOriginal(responseURL),
+	)
+	if err != nil {
+		log.Err(err).Msg("failed to cancel sticker preview")
+		return err
+	}
+
+	return nil
+}
+
+// SendStickerToChannel sends the specified sticker to the Slack channel as a permanent message,
+func (p *Provider) SendStickerToChannel(_ context.Context, userID, channelID, responseURL string, sticker model.StickerBlockActionValue) error {
+	log := p.logger.With().Str(helper.LogStrKeyMethod, "SendStickerToChannel").Logger()
+
+	contextElements := []slack.MixedElement{
+		slack.NewTextBlockObject(slack.MarkdownType, FooterText, false, false),
+		slack.NewImageBlockElement("https://example.com/small-image.png", "alt text for image"),
```

Comment:
what's this for?

Reply:
im using it as a placeholder logo. i could remove it if its not needed
![image](https://github.com/user-attachments/assets/b4a1600b-3835-4b18-b9f8-c2a23b226d55)